sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/models/RedisCache.java

```diff
@@ -519,5 +526,33 @@
          * @return the next stage of Redis Cache definition.
          */
         Update withoutMinimumTlsVersion();
+
+        /**
+         * Update the Redis version.
+         * @param redisVersion the redisVersion value to set
+         * @return the next stage of Redis Cache definition.
+         */
+        Update withRedisVersion(MajorVersion redisVersion);
     }
+
+    /**
+     * major version of redis
+     * for version 4.x.x, use V4
+     * for version 6.x.x, use V6
+     */
+    enum  MajorVersion {
```

Comment:
Is this public?

Reply:
And you need javadoc for all public APIs.

Reply:
I would suggest just call it `RedisVersion`. This will give you some flexibility when service support e.g. "4.1" and you can now add `RedisVersion.V4_1`.